R/user_error_checks.R

```diff
@@ -50,7 +50,11 @@ check_grouped_portfolio_years <- function(
       description = desc,
       immediate = TRUE
     )
-    stop("port_holdings_date contains multiple distinct values", call. = FALSE)
+    # using message and quit, rather than stop(), because server silently
+    # fails if docker container exits with anything other than a 0 status
+    # code
+    message("port_holdings_date contains multiple distinct values")
+    quit(status = 0L)
```

Comment:
might be safer to use `save = "no"` here? I think the default is probably "yes" or "ask"?

actually (emphasis mine)...

> The default is to ask in interactive use but may be overridden by command-line arguments **(which must be supplied in non-interactive use)**.

Reply:
> Some error status values are used by R itself. The default error handler for non-interactive use effectively calls q("no", 1, FALSE) and returns error status 1. Error status 2 is used for R ‘suicide’, that is a catastrophic failure, and other small numbers are used by specific ports for initialization failures. It is recommended that users choose statuses of 10 or more.

so, maybe `quit(save = "no", status = 0, runLast = FALSE)`? your call

I'm not sure why it returns `0` instead of `0L` by default 🤷🏻

Reply:
Good points @cjyetman . I use `0L`, instead of `0`, because I want to make explicit that it's an integer return code.